CefSharp/DevTools/IDevToolsClient.cs

```diff
@@ -20,15 +20,18 @@ public interface IDevToolsClient
         /// </summary>
         event EventHandler<DevToolsEventArgs> DevToolsEvent;
 
+        IDisposable RegisterEventHandler<T>(string eventName, EventHandler<T> eventHandler) where T : DevToolsDomainEventArgsBase;
```

Comment:
I'm thinking we'll change the returned type to https://github.com/cefsharp/CefSharp/blob/master/CefSharp/IRegistration.cs
Keep the code consistent, we can improve the xml doc to provide more detail to the user.